great_expectations/data_context/types/base.py

```diff
@@ -2227,6 +2227,24 @@ def __str__(self) -> str:
         return self.__repr__()
 
 
+class CheckpointValidationConfig(AbstractConfig):
+    def __init__(
+        self, name: Optional[str] = None, id_: Optional[str] = None, **kwargs: dict
+    ) -> None:
+        super().__init__(id_=id_, name=name)
+
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+
+
+class CheckpointValidationConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    name = fields.String(required=False, allow_none=False)
+    id = fields.String(required=False, allow_none=False)
```

Comment:
```suggestion
    id_ = fields.String(required=False, allow_none=False)
```
FYI I ended up using `id_` here due to issues with `instantiate_class_from_config` for the Datasource.

Reply:
Changed!